pubsub/cloud-client/subscriber.py

```diff
@@ -110,6 +110,31 @@ def callback(message):
         time.sleep(60)
 
 
+def listen_for_errors(project, subscription_name):
+    """Receives messages and catches errors from a pull subscription."""
+    subscriber = pubsub_v1.SubscriberClient()
+    subscription_path = subscriber.subscription_path(
+        project, subscription_name)
+
+    def callback(message):
+        print('Received message: {}'.format(message))
+        message.ack()
+
+    subscription = subscriber.subscribe(subscription_path, callback=callback)
+
+    # Blocks the thread while messages are coming in through the stream. Any
+    # exceptions that crop up on the thread will be set on the future.
+    future = subscription.open(callback)
+    try:
+        future.result()
+    except Exception as e:
+        print(
+            'Listening for messages on {} threw an Exception: {}.'.format(
+                subscription_name, e))
+        subscription.close()
```

Comment:
I don't think you *have* to call `subscription.close()`, I'm pretty sure it happens automatically.

Reply:
I'm fine with removing it. Thanks.